Services/DataBaseService/MigrationEngine.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+
+namespace DataBaseService
+{
+    public class MigrationEngine
+    {
+        private readonly IConfiguration configuration;
+
+        public MigrationEngine(IConfiguration configuration)
+        {
+            this.configuration = configuration;
+        }
+
+        /// <summary>
+        /// Migration method. 
+        /// Runs through all the .sql in the directory (configure in the appsettings.json).
+        /// Checks (by its name) if a script was already executed, executes it, 
+        /// and marks as executed (by adding a row to the table).
+        /// </summary>
+        public void Migrate()
+        {
+            var connectionString = configuration.GetConnectionString("MigrationString");
+            var location = configuration.GetSection("Locations")["MigrationScripts"];
+            var scriptsToExecute = Directory.GetFiles(location, "*.sql");
+            var scriptsToWriteDown = new Dictionary<string, string>();
+
+            using (var conn = new SqlConnection(connectionString))
```

Comment:
Create sql transaction, so if any of script is fall everthing will be rollbacked